### ISSUE TITLE
Project dependencies may have API risk issues

### ISSUE THREAD

**Issue by PyDR-Tool** [MedMNIST/MedMNIST#26]
Hi, In **MedMNIST**, inappropriate dependency versioning constraints can cause risks.

Below are the dependencies and version constraints that the project is using

```
numpy
pandas
scikit-learn
scikit-image
tqdm
Pillow
fire
torch
torchvision
```

The version constraint **==** will introduce the risk of dependency conflicts because the scope of dependencies is too strict.
The version constraint **No Upper Bound** and **\*** will introduce the risk of the missing API Error because the latest version of the dependencies may remove some APIs.

After further analysis, in this project,
The version constraint of dependency **pandas** can be changed to *>=0.4.0,<=1.2.5*.
The version constraint of dependency **scikit-learn** can be changed to *>=0.14,<=0.21.3*.
The version constraint of dependency **tqdm** can be changed to *>=4.36.0,<=4.64.0*.
The version constraint of dependency **Pillow** can be changed to *==9.2.0*.
The version constraint of dependency **Pillow** can be changed to *>=2.0.0,<=9.1.1*.

The above modification suggestions can reduce the dependency conflicts as much as possible, 
and introduce the latest version as much as possible without calling Error in the projects.

The invocation of the current project includes all the following methods.

<details><summary>The calling methods from the pandas</summary>
 <pre>pandas.read_csv
</pre>
</details>
<details><summary>The calling methods from the scikit-learn</summary>
 <pre>sklearn.metrics.accuracy_score
sklearn.metrics.roc_auc_score
</pre>
</details>
<details><summary>The calling methods from the tqdm</summary>
 <pre>tqdm.trange
</pre>
</details>
<details><summary>The calling methods from the Pillow</summary>
 <pre>PIL.Image.fromarray
</pre>
</details>
<details><summary>The calling methods from the all methods</summary>
 <pre>RuntimeError
numpy.random.rand.sum
fire.Fire
next
format
numpy.stack
ys.append
save_fn
setuptools.setup
numpy.random.rand
list
filename.split
available
medmnist.Evaluator.get_dummy_prediction
f.read
os.path.join
zip
time.time
download
os.path.exists
self.download
medmnist.utils.montage3d
df.append.sort_index
medmnist.utils.montage2d
frames.append
filename.split.split
save
split_.startswith
join
cls.evaluate
self.labels.max
key.INFO.medmnist.getattr
shuffle_iterator
self.get_standard_evaluation_filename
map
warnings.DeprecationWarning
medmnist.info.INFO.keys
pandas.DataFrame
index.self.labels.astype
get_default_root
y_score.pd.DataFrame.to_csv
key.INFO.medmnist.getattr.montage
numpy.argmax
key.INFO.medmnist.getattr.save
flag.INFO.medmnist.getattr
key.endswith
y_true.squeeze.squeeze
os.path.split
glob.glob
Metrics
pandas.read_csv
medmnist.utils.montage2d.save
self.__len__
pprint.pprint
open.close
df.append.append
medmnist.utils.save2d
medmnist.Evaluator.parse_and_evaluate
self.transform.convert
medmnist.Evaluator
os.path.expanduser
getAUC
xs.append
readme
range
setuptools.find_packages
dataset._collate_fn
open
info
self.__len__.append
path.endswith
sklearn.metrics.accuracy_score
y_score.squeeze.squeeze
sklearn.metrics.roc_auc_score
medmnist.utils.save_frames_as_gif
data.append
open.write
montage2d
os.makedirs
cls
getACC
numpy.load
random.shuffle
tqdm.trange
torchvision.datasets.utils.download_url
load_fn.save
os.remove
print
getattr
load_fn
medmnist.utils.save3d
skimage.util.montage
montage_frames.append
self.transform
self.target_transform
medmnist.Evaluator.evaluate
df.append.to_csv
len
numpy.random.choice
frames.save
PIL.Image.fromarray
numpy.array
collections.namedtuple
i.y_true.astype
warnings.warn
idx.append
</pre>
</details>

@developer
Could please help me check this issue?
May I pull a request to fix it?
Thank you very much.
